lib/facade_converter.ts

```diff
@@ -354,6 +377,73 @@ export class FacadeConverter extends base.TranspilerBase {
     },
   };
 
+  private es6Promises: ts.Map<CallHandler> = {
+    'Promise.resolve': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.value(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise.reject': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.error(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise.catch': (c: ts.CallExpression, context: ts.Expression) => {
+      this.visit(context);
+      this.emit('.catchError(');
+      this.visitList(c.arguments);
+      this.emit(')');
+    },
+    'Promise': (c: ts.CallExpression, context: ts.Expression) => {
+      if (c.kind != ts.SyntaxKind.NewExpression) {
+        return true;
+      }
+      this.assert(c, c.arguments.length == 1, 'Promise construction must take 2 arguments.');
+      this.assert(
+          c, c.arguments[0].kind == ts.SyntaxKind.ArrowFunction ||
+              c.arguments[0].kind == ts.SyntaxKind.FunctionExpression,
+          'Promise argument must be a function expression (or arrow function).');
+      let callback: ts.FunctionLikeDeclaration;
+      if (c.arguments[0].kind == ts.SyntaxKind.ArrowFunction)
+        callback = <ts.FunctionLikeDeclaration>(<ts.ArrowFunction>c.arguments[0]);
+      if (c.arguments[0].kind == ts.SyntaxKind.FunctionExpression)
+        callback = <ts.FunctionLikeDeclaration>(<ts.FunctionExpression>c.arguments[0]);
+      this.assert(
+          c, callback.parameters.length > 0 && callback.parameters.length < 3,
+          'Promise executor must take 1 or 2 arguments (resolve and reject).');
+
+      this.assert(
+          c, callback.parameters[0].name.kind == ts.SyntaxKind.Identifier,
+          'First argument of the Promise executor is not a straight parameter.');
+      let resolveParameterIdent = <ts.Identifier>(callback.parameters[0].name);
+      this.assert(
+          c, resolveParameterIdent.text != '_completer',
+          'Promise executor arguments cannot have "_completer" as a name.');
+
+      this.emit('(() {');  // Create a new scope.
```

Comment:
AFAIK dart2js does not inline IIFEs.

Can we do away with the extra scope, assuming the completer var name is unique enough?